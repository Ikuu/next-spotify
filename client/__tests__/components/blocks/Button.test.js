import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from '../../../components/blocks/Button';

test('Button renders', () => {
  const tree = create(<Button text="Text" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button renders with aria-label', () => {
  const tree = create(<Button ariaLabel="Different text" text="Text" />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Button handles click', () => {
  const mockOnClick = jest.fn();
  const wrapper = shallow(<Button text="Test" onClick={mockOnClick} />);
  wrapper.find('button').simulate('click');
  expect(mockOnClick).toBeCalled();
});
