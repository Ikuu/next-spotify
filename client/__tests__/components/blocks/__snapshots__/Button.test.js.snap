// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders 1`] = `
<button
  aria-label="Text"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "color": "black",
      "cursor": "pointer",
    }
  }
  type="button"
>
  Text
</button>
`;

exports[`Button renders with aria-label 1`] = `
<button
  aria-label="Different text"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "color": "black",
      "cursor": "pointer",
    }
  }
  type="button"
>
  Text
</button>
`;
